mint-auditor/src/db/models/gnosis_safe_deposit.rs

```diff
@@ -141,11 +141,13 @@ impl GnosisSafeDeposit {
 
     /// Attempt to find a [GnosisSafeDeposit] that has a given nonce and no
     /// matching entry in the `audited_mints` table.
-    pub fn find_unaudited_deposit_by_nonce(
+    pub fn find_unaudited_deposit_by_nonce_and_token_address(
```

Comment:
As mentioned on a previous comment, I don't think we can encounter the same nonce twice on the Gnosis side, since it is derived from the ethereum tx hash - which is guaranteed to be unique.

Does that make sense? Please double check this assumption :)